interleavebench/main.go

```diff
@@ -0,0 +1,324 @@
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"os"
+	"os/exec"
+	"os/signal"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	_ "github.com/lib/pq"
+)
+
+var dbName = "roachbench"
+
+var concurrency = flag.Int("concurrency", 2*runtime.NumCPU(), "Number of concurrent writers inserting blocks")
+var duration = flag.Duration("duration", 0, "The duration to run. If 0, run forever.")
+
+// For postgres, use something like "postgres://$(whoami)@locahost:5432/template1?sslmode=disable
+var dbURL = flag.String("url", "", "Database URL.")
+
+// Data loading and dumping flags.
+var drop = flag.Bool("drop", false, "Drop and recreate the database.")
+var load = flag.Bool("load", false, "Generate fresh data from --seed (deterministic). Use with --drop.")
+var dumpfile = flag.String("dump-file", "", "If specified, will dump DB contents to this file.")
+var loadfile = flag.String("load-file", "", "Filepath of file generated from --dump to load into database.")
+var seed = flag.Int64("seed", 42, "Pseudo-random seed used to generate data.")
+var dbms = flag.String("dbms", "cockroach", "DBMS used (values: 'cockroach' or 'postgres'). Relevant for --load and --dump.")
+var databases = map[string]bool{
+	"cockroach": true,
+	"postgres":  true,
+}
+var variant = flag.String("variant", "normal", "How tables are created (values: 'normal' or 'interleaved'). Use with --load/--load-file.")
+var variants = map[string]bool{
+	"normal":      true,
+	"interleaved": true,
+}
+
+var nMerchants = flag.Int("merchants", 10000, "Number of rows in table <merchant> to generate. Use with --load.")
+var nProducts = flag.Int("products", 1000000, "number of rows in table <product> to generate. Use with --load.")
+var nVariants = flag.Int("variants", 10000000, "number of rows in table <variant> to generate. Use with --load.")
+var nStores = flag.Int("stores", 50000, "Number of rows in table <store> to generate. Use with --load.")
+
+var schemaVar schemaVariant
+var tables []tableName
+
+func init() {
+	flag.Parse()
+
+	// Schema DDL statements initialization.
+	initDDL()
+
+	if !databases[*dbms] {
+		log.Fatalf("--database must either be 'cockroach' or 'postgres'")
+	}
+	if !variants[*variant] {
+		log.Fatalf("--variant must either be 'normal' or 'interleaved'")
+	}
+
+	if *dbms == "postgres" && *variant != "normal" {
+		log.Fatalf("--database=postgres only works with --variant=normal")
+	}
+
+	schemaVar = schemaVariant(*variant)
+	tables = variantTables[schemaVar]
+
+	dbName = dbName + "_" + *variant
+
+	if *dbURL == "" {
+		// Need to connect to template DB to perform drops and loads.
+		if *dbms == "postgres" && (*drop || *load || *loadfile != "") {
+			*dbURL = defaultURL("template1")
+		} else {
+			*dbURL = defaultURL(dbName)
+		}
+
+	}
+}
+
+var stderr = log.New(os.Stderr, "", 0)
+
+func defaultURL(name string) string {
+	if *dbms == "postgres" {
+		return fmt.Sprintf("postgres://richardwu@localhost:5432/%s?sslmode=disable", name)
```

Comment:
Do you need to include your username here?

Reply:
Also, the usual practice for load generators it to take a URL on the command line rather than a flag. This allows you to point the generator at a remote node.

Reply:
Changed this to retrieve the current user and also take an optional CL argument for the URL.